lingua_franca/lang/parse_en.py

```diff
@@ -882,12 +918,14 @@ def date_found():
                 start -= 1
                 # parse 15 of July, June 20th, Feb 18, 19 of February
         elif word in months or word in monthsShort and not fromFlag:
-            try:
-                m = months.index(word)
-            except ValueError:
-                m = monthsShort.index(word)
+            m = month_to_int_en(word)
+
+            if wordPrev in ["next", "this", "last", "past", "previous"]:
```

Comment:
Looks like "this <week/month/year>" still returns nothing. Idk if that's in scope.